Feature/multitenant

INovozhilov opened the pull request:
Подключение multitenant

INovozhilov committed:
Подключение библиотеки

INovozhilov committed:
Merge remote-tracking branch 'remotes/origin/master' into feature/multitenant

INovozhilov committed:
Подключение основных объектов

INovozhilov committed:
Проверка на eventclass/create/

INovozhilov commented on crm/middleware.py:
```diff
@@ -0,0 +1,29 @@
+
+
+def set_current_tenant_for_user(current_user):
+    if current_user.is_manager:
```
Мне не нравится это место. Вероятно нужно унаследовать весь user так, чтобы в нем была связь с компанией.
Но тогда непонятно что делать со staff.
Если делать nullable связь, то будет потенциальная дыра.
Может админа привязывать к 1-й, служебной компании?

ciklop replied:
Честно говоря не вижу особой проблемы. Можно же сделать как - если пользователь не менеджер, не тренер, и не staff - ты выкидывать его на мороз с 403.
А вот как такие пользователи могут появится - ну сходу приходит только ковыряние руками в админке, что уже не правильно. Так-что сей подход нормальный я считаю.

Только не совсем понимаю зачем было выносить в отдельную функицю, а не в рамках класса мидлвари.
Ну и не хватает проверки не тренера.

ciklop commented on sportcrm/settings/production.py:
```diff
@@ -12,4 +12,7 @@
-    conn_max_age=600, ssl_require=True)
+    conn_max_age=600, ssl_require=True,
+    # django-multitenant db engine for foreign keys
+    engine='libs.django_multitenant.django_multitenant.backends.postgresql')
```
Нормальная там код, они там просто несколько хаков делают чтоб на лету многоголовость работала. Ничего критичного. Другое дело что с sqlite3 может и не подружиться эта мултитеннатность.

ciklop commented on crm/models.py:
```diff
@@ -63,7 +67,28 @@ def vk_data(self, data_key: str) -> Optional[str]:
 
 
-class Location(models.Model):
+class Company(models.Model):
```
И как я понимаю - не все модели ещё перенесены

INovozhilov committed:
Подключение библиотеки через requirements

INovozhilov committed:
Перевод всех моделей, уникальный ключ, фильтры в формах

INovozhilov committed:
Изменен бэкэнд

ciklop committed:
Merge branch 'master' into feature/multitenant

# Conflicts:
#	crm/models.py

ciklop committed:
Update of multitenant usage

User now is connected with company. Company must be set during user creation. If there no company provided, default 'INTERNAL' will be used.
Internal company is created during migration.
Company name act now as uuid, for internal uses. For client display_name is user.

ciklop committed:
Update prod config

ciklop committed:
Fix test for multitenant support

ciklop committed:
Add basic view test functions

ciklop committed:
Clean db engine in prod
